### ISSUE TITLE
Tipping the bar tender

### ISSUE THREAD

**Issue by rootzoll** [arcbtc/M5StackSats#19]
see https://twitter.com/provoost/status/1206204457127727104?s=20

A lightning tipping-jar could be a static LNURL QR code, a little m5stack (without the keyboard) or a ESP32 with a small epaper ... showing always a valid invoice where you are free to set the amount. Would be great if on the backend the tipping jar can get connected to telegram (or alike) so that a small notification is send to bar tender about tip.

**Comment by Sjors** [comment]
It would be useful to ask around in bars and restaurants how they generally handle tipping with electronic payments.

From a customer perspective having to make two separate payments, one to pay the invoice and one for the tipping jar, seems annoying, but not terrible.

Having two separate devices for it seems too complicated.

**Comment by rootzoll** [comment]
I think for a lot of customers tipping seperately is socially more relaxed. Because in the room77 the bar tender is handling the POS ... punching in the amount to pay and would also have to ask the customer with a built-in tipping feature "how much do you want to tip me?" and you have to answere. This forces a social pressure moment. Having a seperate tipping jar removes having this mandatory and makes it a more relaxed voluntary act. The bar tender can still say "and if you like to tip me .. there is the tipping jar you can use afterwards" to raise awareness, but thats a more relaxed indirect way and not mandatory.

Another way would be overpaying a lightning invoice .. but I think wallets are to fragmented to make that process smoothly.

**Comment by Sjors** [comment]
> bar tender is handling the POS ... punching in the amount to pay and would also have to ask the customer with a built-in tipping feature "how much do you want to tip me?"

Ideally they would type the invoice amount, which can't be undone, and then give the device to the customer who can only add a tip, but not change the amount.

**Comment by rootzoll** [comment]
> Ideally they would type the invoice amount, which can't be undone, and then give the device to the customer who can only add a tip, but not change the amount.

Ah OK. So while the customer is presented the POS to scan the invoice the customer can optional press a Button with the dynamic lable "Add TIP" and enter an additional amount in EUR. The POS renews the invoice then and lets the customer pays it. All is one LN transaction - the tip amount should be part of the invoice description so the node (admin) can calculate and payout all tips to the bar tender.

That could be a way to do it. Practically within the Room77 setup this would mean that the individual tip will be hard to match to a unique bar tender (without adding further complexity) ... but all tips a day/week/month could be split by some fair rule between all bar personal.

**Comment by Sjors** [comment]
Something like that, but again it's useful to actually talk to bars / restaurants and see how they handle tips now with electronic payments.

**Comment by arcbtc** [comment]
I have noticed the gratuity option in PoS terminals in the UK has been phased out, because of that awkwardness Christian mentioned. In the LNTXBOT version of M5StackSats I did add the ability to press the middle button to generate a "0" invoice, which could have a "tip *timestamp*" memo. Physically writing "tip" on the button, could encourage but not pressure tipping.
This is an ideal scenario for LNURL-pay, which more wallets are starting to support, and wouldn't require additional hardware. Each barkeep could just have a printed QR on the bar, or on their person
